npep/npep-137-conformance-profiles.md

```diff
@@ -0,0 +1,187 @@
+# NPEP-137: Conformance Profiles
+
+* NOTE: Adapted with love from https://gateway-api.sigs.k8s.io/geps/gep-1709/ (Then
+a question may arise, why do we need our own NPEP? Answer: Gateway API Project has
+lots of resources and features and it gets very complicated to adopt their same API.
+We need to tune the profiles to be compatible with what works for Network Policy API
+project, thus the need for this NPEP). Our profiles are much simpler than what GatewayAPI
+supports
+* Issue: [#137](https://github.com/kubernetes-sigs/network-policy-api/issues/137)
+* Status: Implementable
+
+## TLDR
+
+Add conformance profiles for existing conformance tests which implementations
+can leverage when running the conformance test suite in their downstream repos
+to report the results of the tests back to the Network Policy API project and
+receive recognition (eg. "conformance badge")
+
+## Goals
+
+* Add conformance profiles for existing tests that downstream implementations
+can subscribe to to run tests associated to the supported API feature sets
+
+* Add a reporting mechanism via a new CRD where conformance results can be
+reported back to the Network Policy API project and provide "conformance badges"
+via Network Policy official website to recognize these implementations.
+
+* Expand existing conformance testing framework to account for profiles
+
+## Non-Goals
+
+* The first iteration will just have a simple repo under which the report
+results will be stored, no automated reporting infrastructure will be built for this now.
+Implementations will need to open PRs against network-policy-api repo to upload
+the results.
+
+## Introduction
+
+Currently, the conformance tests are grouped into `StandardCoreFeatures` and
```

Comment:
nit for another PR, can we just make this `CoreFeatures` and `ExtendedFeatures`?

Reply:
i actually like this suggestion, I will fix this in this NPEP and also open a new PR to fix the naming in our existing tests..

Reply:
@astoycos : See https://github.com/kubernetes-sigs/network-policy-api/pull/155

Reply:
Considering how you have crd/experimental already maybe we call them "ExperimentalFeatures"